libethereum/BlockChain.cpp

```diff
@@ -343,10 +354,10 @@ void BlockChain::rebuild(fs::path const& _path, std::function<void(unsigned, uns
 
     // Keep extras DB around, but under a temp name
     m_extrasDB.reset();
-    fs::rename(extrasPath / fs::path("extras"), extrasPath / fs::path("extras.old"));
-    std::unique_ptr<db::DatabaseFace> oldExtrasDB(
-        new db::DBImpl(extrasPath / fs::path("extras.old")));
-    m_extrasDB.reset(new db::DBImpl(extrasPath / fs::path("extras")));
+    if (!db::isMemoryDB())
```

Comment:
I think this `rebuild()` method doesn't make any sense for MemoryDB (it rebuilds the state DB by replaying all blocks from blockchain DB), so let's just return immediately in the beginning if it's MemoryDB (maybe output some log message, too)

Reply:
I took a look at when rebuild is called and it's only called when re-opening an existing database, which obviously doesn't make sense when running MemoryDB. I'll go ahead and return immediately.